syntax = "proto3";

package protobuf;

service TodoService {
  // Add a user
  rpc AddUser(User) returns (User) {}

  // Add, update ToDo
  rpc AddTodo(ToDo) returns (ToDo) {}
  rpc UpdateTodo(ToDo) returns (ToDo) {}

  // Get or Delete list of Todos
  rpc GetToDos(User) returns (stream ToDo) {}
  rpc DeleteToDos(User) returns (stream ToDo) {}
}

message User {
  int32 id = 1;
  string name = 2;
}

message ToDo {
  int32 id = 1;
  User user = 2;
  string text = 3;
  bool is_done = 4;
}
